modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -612,223 +622,191 @@ private CompletableFuture<Void> onTableDelete(DropTableEventParameters parameter
     /**
      * Updates or creates partition raft groups and storages.
      *
-     * @param causalityToken Causality token.
      * @param assignmentsFuture Table assignments.
      * @param table Initialized table entity.
      * @return future, which will be completed when the partitions creations done.
      */
-    private CompletableFuture<?> createTablePartitionsLocally(
-            long causalityToken,
+    private CompletableFuture<Void> startLocalPartitionsAndUpdateClients(
```

Comment:
Where do we actually update clients? And where do we actually use that updated clients?

Reply:
Inside of startPartitionAndStartClient, in thenComposeAsync for startGroupFut. After that they are put into InternalTableImpl
`((InternalTableImpl) internalTbl).updateInternalTableRaftGroupService(partId, updatedRaftGroupService);`

Reply:
![NOTSUREIF](https://github.com/apache/ignite-3/assets/18287137/a878a7e2-73d6-4575-9e7f-b831b5f7b475)

Reply:
Makes sense.